test/demo/createCenter.inttest.ts

```diff
@@ -0,0 +1,204 @@
+import {assertNotificationStored, getAppNotifications} from './utils/storageRemote';
+import {delay} from './utils/delay';
+import {getToastWindow} from './utils/toastUtils';
+import {assertDOMMatches, CardType} from './utils/noteCardUtils';
+
+
+const testManagerIdentity = {uuid: 'test-app', name: 'test-app'};
+
+describe('When creating a notification with the center showing', () => {
+    let testApp: Application;
+    let testWindow: Window;
+
+    beforeAll(async () => {
+        // Toggle the center on/off based on test type
+        if (!(await isCenterShowing())) {
+            await notifsRemote.toggleNotificationCenter(testManagerIdentity);
+        }
+    });
+
+    beforeEach(async () => {
+        testApp = await createApp(testManagerIdentity, {});
+        testWindow = await testApp.getWindow();
+    });
+
+    afterEach(async () => {
+        await notifsRemote.clearAll(testWindow.identity);
+        await testApp.quit();
+    });
+
+    describe('When passing a valid set of options', () => {
+        const options: NotificationOptions = {
+            body: 'Test Notification Body',
+            title: 'Test Notificaiton Title',
+            id: 'test-notification-0',
+            icon: 'https://openfin.co/favicon.ico'
+        };
+
+        let createPromise: Promise<Notification>;
+        let note: Notification;
+        beforeEach(async () => {
+            createPromise = notifsRemote.create(testWindow.identity, options);
+
+            // We want to be sure the operation is completed, but don't care if it succeeds
+            note = await createPromise.catch(() => ({} as Notification));
+        });
+
+        test('The promise resolves to the fully hydrated notification object', async () => {
+            await expect(createPromise).resolves;
+            expect(note).toMatchObject(options);
+        });
+
+        test('One card appears in the notification center', async () => {
+            const noteCards = await getCenterCardsByNotification(testApp.identity.uuid, note.id);
+            expect(noteCards).toHaveLength(1);
+        });
+
+        test('The card has the same data as the returned notification object', async () => {
+            await assertDOMMatches(CardType.CENTER, testApp.identity.uuid, note);
+        });
+
+        test('The notification is added to the persistence store', async () => {
+            await assertNotificationStored(testWindow.identity, note);
+        });
+
+        test('The notification is included in the result of a getAll call', async () => {
+            const appNotes = await notifsRemote.getAll(testWindow.identity);
+            expect(appNotes).toContainEqual(note);
+        });
+
+        test('No toast is shown for the created notification', async () => {
+            // The notification is created immediately before this, so we need
+            // a slight delay to allow time for the toast to spawn.
+            await delay(100);
```

Comment:
Worth pulling out a `toastSpawnDelay` constant?

Reply:
You ok to have a quick offline chat about all the delay comments at once.